### PR TITLE
Fixes to Receiving SMS Replies

### PR DESCRIPTION
**Pull request by dkobia**
- Since we're saving phone numbers as 'numeric' in the database we need to remove non-numeric characters before comparing any phone numbers
- We're also using strrpos to compare 'sender' and 'from' because some people might use '000-000-0000' in their sms config vs '+10000000000'. Both work. **strrpos** allows us to compare two numbers in case one doesn't have the country code. Twilio is obviously US biased.
- Append the '+' in the Twilio send() as the phone number is stored as numerical for consistency
- Reverted pings status column to 'pending/received'

@ekala -- need a review

### CODE REVIEW

File: application/classes/Controller/Sms/Twilio.php [hunk 1]
```diff
@@ -39,16 +41,16 @@ public function action_reply()
 			
 			// Use the last id of the ping to tag the pong
 			// TODO: Review
-			$ping = DB::query(array(DB::expr('COUNT(id)'), 'ping_id'))
+			$ping = DB::select(array(DB::expr('COUNT(id)'), 'ping_id'))
```
Comment (ekala) [inline application/classes/Controller/Sms/Twilio.php]: @dkobia  Should be `DB::expr('MAX('id'))`

File: application/classes/Controller/Sms/Twilio.php [hunk 2]
```diff
@@ -39,16 +41,16 @@ public function action_reply()
...
 				->where('contact_id', '=', $contact->id)
 				->where('type', '=', 'phone')
-				->where('status', 'pending')
+				->where('status', '=', 'pending')
 				->execute()
 				->as_array();
 			
 			// Record the pong
-			if ( ! count($ping))
+			if ( count($ping) > 0 )
```
Comment (dkobia) [inline application/classes/Controller/Sms/Twilio.php]: not sure why this was returning false in some instances for a count of greater than 0 when using:

```
if ( ! count($ping) )
```

File: application/classes/Controller/Sms/Twilio.php (hunk 1 above)
Comment (dkobia) [inline application/classes/Controller/Sms/Twilio.php]: Fixing

**Comment by ekala** [comment]
LGTM. :+1: for merge

**Pull request merged**